src/main/java/com/microfocus/octane/gitlab/services/GitlabServices.java

```diff
@@ -112,53 +103,70 @@ private void init() throws MalformedURLException {
         executor.scheduleAtFixedRate(new TestResultsCleanUpRunnable(applicationSettings.getConfig().getTestResultsOutputFolderPath()),  TestResultsCleanUpRunnable.INTERVAL, TestResultsCleanUpRunnable.INTERVAL, TimeUnit.MINUTES);
     }
 
-    public static String generateNewToken() {
+    private Boolean addWebHookToProject(Object projectId, boolean deleteOldWebHook) throws GitLabApiException {
```

Comment:
The returns value is always true and it is never used, should be removed.

The behavior of this method should be to always replace existing hooks (0...n) (you should never be in the case when there are multiple hooks, but it's good that is is handled) with a new one (1). Just adding a new hook without deleting the existing ones shouldn't be permitted.